### PR TITLE
Fix/suppress swig warnings

### PR DESCRIPTION
**Pull request by ahmed-agiza**
- Suppress all SWIG warnings jungle.
- Fix if guard indentation causing a warning in lefin.
- Remove unused struct timeval in logger.
- Update Jenkins docker run command (no root permissions/clean issues).

### CODE REVIEW

File: src/lefin/lefin.cpp [hunk 1]
```diff
@@ -577,9 +577,10 @@ void lefin::layer( lefiLayer * layer )
 
     dbTechLayerType type( dbTechLayerType::ROUTING );
 
-    if (layer->hasType())
+    if (layer->hasType()) {
```
Comment (jjcherry56) [inline src/lefin/lefin.cpp]: how is this relevant to suppressing swig errors?
I personally don't agree with this formatting style but I think it is best to not randomly introduce changes to the original unless they are universally applied.

**Comment by jjcherry56** [comment]
Although I applaud suppressing the incredibly noisy swig warnings, I think the bigger issue is the overzealous inclusion of header files in the swig world. What about using #swig_ignore or something to "comment" out sections of the headers that shouldn't be there? I have a pending branch to remove large chunks of the existing includes because they are largely unusable from tcl/swig because they are too c++ specific.

File: src/lefin/lefin.cpp (hunk 1 above)
Comment (ahmed-agiza) [reply src/lefin/lefin.cpp]: It was just causing a **Wmisleading-indentation** warning, I reverted it back to the old style and adjusted the indentation instead.

**Comment by ahmed-agiza** [comment]
> Although I applaud suppressing the incredibly noisy swig warnings, I think the bigger issue is the overzealous inclusion of header files in the swig world. What about using #swig_ignore or something to "comment" out sections of the headers that shouldn't be there? I have a pending branch to remove large chunks of the existing includes because they are largely unusable from tcl/swig because they are too c++ specific.

I agree, there is huge clean up that needs to be done for the SWIG headers, there is also a big list of warnings that can be resolved by renaming operators or adding typecheck typemaps, I just felt they are polluting the logs all the time unnecessarily with barely anyone looking at them, so it might be better to suppress them for now until the clean up is done. Feel free to close the PR if you prefer to work on the headers first.
